File: Server/src/main/java/com/example/Server/model/Transaction.java
package com.example.Server.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@Getter
@Setter
@Table(name = "transaction")
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String title;

    @Column
    private String type;

    @Column
    private String category;

    @Column
    private double amount;

    @Column
    @JsonFormat(pattern = "MMM d, yyyy hh:mm:ss a")
    private Date timestamp;

}

File: native-localDb-server/app/src/main/java/com/example/expensewizard/model/Transaction.kt
package com.example.expensewizard.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date
@Entity
data class Transaction(
   @PrimaryKey(autoGenerate = true)
    var id: Long,
    var title: String,
    var type:String,
    var category: String,
    var amount: Double,
    var timestamp: Date,
) {
//    companion object {
//        var currentId = 1
//    }
//
//    constructor(title: String, type: Type, category: String, amount: Double, timestamp: Date) :
//            this(currentId++,title,type,category,amount,timestamp)
//    {
//        Log.i("Model Transaction Class: ", "Current id is $currentId") }


}